aws-android-sdk-core/src/main/java/com/amazonaws/http/UrlHttpClient.java

```diff
@@ -85,6 +94,38 @@ public HttpResponse execute(final HttpRequest request) throws IOException {
         return createHttpResponse(request, connection);
     }
 
+    HttpURLConnection getConnection(URL url) throws IOException {
+        if (config.getProxyHost() != null) {
+            SocketAddress socketAddress;
+            if (config.getProxyPort() != -1) {
+                socketAddress = new InetSocketAddress(config.getProxyHost(), config.getProxyPort());
+            } else {
+                socketAddress = new InetSocketAddress(config.getProxyHost(), 80);
+            }
+            proxy = new Proxy(Proxy.Type.HTTP, socketAddress);
+
+            if (config.getProxyUsername() != null && config.getProxyPassword() != null) {
+                Authenticator authenticator = new Authenticator() {
+                    @Override
+                    protected PasswordAuthentication getPasswordAuthentication() {
+                        return (new PasswordAuthentication(config.getProxyUsername(),
```

Comment:
We should probably not hold onto the `config` object inside this object. See [other use](https://github.com/aws-amplify/aws-sdk-android/blob/3bcbc844d5498f67d01e0c66be87c90a17b03f1f/aws-android-sdk-core/src/main/java/com/amazonaws/http/HttpClientFactory.java#L114)